examples/dts/sub-orchestrations-with-fan-out-fan-in/README.md

```diff
@@ -0,0 +1,102 @@
+# Portable SDK Sample for Sub Orchestrations and Fan-out / Fan-in
+
+This sample demonstrates how to use the Durable Task SDK, also known as the Portable SDK, with the Durable Task Scheduler to create orchestrations. These orchestrations not only spin off child orchestrations but also perform parallel processing by leveraging the fan-out/fan-in application pattern.
+
+The scenario showcases an order processing system where orders are processed in batches. 
+
+> Note, for simplicity, this code is contained within a single source file. In real practice, you would have 
+
+# Prerequisites
+ - [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli)
+ - [`az durabletask` CLI extension](https://learn.microsoft.com/en-us/cli/azure/durabletask?view=azure-cli-latest)
+
+## Running the Examples
+There are two separate ways to run an example:
+
+- Using the Emulator
+- Using a deployed Scheduler and Taskhub
+
+### Running with a Deployed Scheduler and Taskhub rResource
+1. To create a taskhub, follow these steps using the Azure CLI commands:
+
+Create a Scheduler:
+```bash
+az durabletask scheduler create --resource-group --name --location --ip-allowlist "[0.0.0.0/0]" --sku-capacity 1 --sku-name "Dedicated" --tags "{'myattribute':'myvalue'}"
+```
+
+Create Your Taskhub:
+```bash
+az durabletask taskhub create --resource-group <testrg> --scheduler-name <testscheduler> --name <testtaskhub>
+```
+
+2. Retrieve the Endpoint for the Scheduler: Locate the taskhub in the Azure portal to find the endpoint.
+
+3. Set the Environment Variables:
+Bash:
+```bash
+export TASKHUB=<taskhubname>
+export ENDPOINT=<taskhubEndpoint>
+```
+Powershell:
+```powershell
+$env:TASKHUB = "<taskhubname>"
+$env:ENDPOINT = "<taskhubEndpoint>"
+```
+
+4. Install the Correct Packages
+```bash
+pip install -r requirements.txt
+```
+
+4. Grant your developer credentials the `Durable Task Data Contributor` Role.
+
+### Running with the Emulator
+The emulator simulates a scheduler and taskhub, packaged into an easy-to-use Docker container. For these steps, it is assumed that you are using port 8080.
+
+1. Install Docker: If it is not already installed.
+
+2. Pull the Docker Image for the Emulator:
+```bash
+docker pull mcr.microsoft.com/dts/dts-emulator:v0.0.6
+```
+
+3. Run the Emulator: Wait a few seconds for the container to be ready.
+```bash
+docker run --name dtsemulator -d -p 8080:8080 mcr.microsoft.com/dts/dts-emulator:v0.0.4
+```
+
+3. Set the Environment Variables:
```

Comment:
Let's remove environment variables from the local emulator setup instructions. The sample code can assume `localhost` and `default` if these variables don't exist. This will save users a lot of time and effort when getting started.